src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1028,6 +1028,22 @@ public IEqualityComparer<T> Comparer
             }
         }
 
+        /// <summary>
+        /// Ensures that the hashset can hold up to 'capacity' entries without any further expansion of its backing storage
+        /// </summary>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+                throw new ArgumentOutOfRangeException(nameof(capacity));
+            if (_slots != null && _slots.Length >= capacity)
+                return _slots.Length;
+            if (_buckets == null)
```

Comment:
If_buckets!=null then_slots!= null. So you can eliminate the null check on slots by putting the buckets check first. Then the normal case of not null had one fewer comparison